src/dtx/dtx_srv.c

```diff
@@ -148,7 +169,16 @@ dtx_handler(crt_rpc_t *rpc)
 
 			i += count;
 		}
+
+		d_tm_inc_counter(dpm->dpm_batched_total,
+				 din->di_dtx_array.ca_count);
+		d_tm_get_counter_srv(dpm->dpm_batched_total, &ent_cnt);
+		d_tm_get_counter_srv(dpm->dpm_total[opc], &opc_cnt);
+		d_tm_set_counter(dpm->dpm_batched_average,
+				 ent_cnt / (opc_cnt + 1));
```

Comment:
It would be better to take advantage of the D_TM_STATS_GAUGE metrics type, instead of having two separate metrics for total and average. If you make the total a stats gauge, the control plane will present the average (along with some other stats which are probably less useful for this case).

Reply:
In fact, what we care about is the average DTX entries per DTX commit RPC, that is NOT the average count of total committed DTX entries. We can only maintain the batched_total, and the user can calculate the average DTX entries via "batched_total / commit_RPC_count", but I am not sure whether it is convenient for the user or not.

Reply:
Ah, I follow. I think a different name may be clearer, maybe something like "average batch size"? I think it's fine to include it.

Reply:
How about "batched_degree"? Is it clear enough?

Reply:
Fine with me. Thanks.